deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/model/Darknet19.java

```diff
@@ -0,0 +1,220 @@
+package org.deeplearning4j.zoo.model;
+
+import lombok.NoArgsConstructor;
+import org.deeplearning4j.nn.api.Model;
+import org.deeplearning4j.nn.api.OptimizationAlgorithm;
+import org.deeplearning4j.nn.conf.ComputationGraphConfiguration;
+import org.deeplearning4j.nn.conf.ComputationGraphConfiguration.GraphBuilder;
+import org.deeplearning4j.nn.conf.ConvolutionMode;
+import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
+import org.deeplearning4j.nn.conf.WorkspaceMode;
+import org.deeplearning4j.nn.conf.inputs.InputType;
+import org.deeplearning4j.nn.conf.layers.*;
+import org.deeplearning4j.nn.graph.ComputationGraph;
+import org.deeplearning4j.nn.weights.WeightInit;
+import org.deeplearning4j.zoo.ModelMetaData;
+import org.deeplearning4j.zoo.PretrainedType;
+import org.deeplearning4j.zoo.ZooModel;
+import org.deeplearning4j.zoo.ZooType;
+import org.nd4j.linalg.activations.Activation;
+import org.nd4j.linalg.activations.impl.ActivationLReLU;
+import org.nd4j.linalg.learning.config.Nesterovs;
+import org.nd4j.linalg.lossfunctions.LossFunctions;
+
+/**
+ * Darknet19
+ *  Reference: https://arxiv.org/pdf/1612.08242.pdf
+ *
+ * <p>ImageNet weights for this model are available and have been converted from https://pjreddie.com/darknet/imagenet/
+ * using https://github.com/allanzelener/YAD2K .</p>
+ *
+ * There are 2 pretrained models, one for 224x224 images and one fine-tuned for 448x448 images.
+ * Call setInputShape() with either {3, 224, 224} or {3, 448, 448} before initialization.
+ * The channels of the input images need to be in RGB order (not BGR), with values normalized within [0, 1].
+ * The output labels are as per https://github.com/pjreddie/darknet/blob/master/data/imagenet.shortnames.list .
+ *
+ * @author saudet
+ */
+@NoArgsConstructor
+public class Darknet19 extends ZooModel {
+
+    private int[] inputShape = {3, 224, 224};
+    private int numLabels;
+    private long seed;
+    private int iterations;
+    private WorkspaceMode workspaceMode;
+    private ConvolutionLayer.AlgoMode cudnnAlgoMode;
+
+    public Darknet19(int numLabels, long seed, int iterations) {
+        this(numLabels, seed, iterations, WorkspaceMode.SEPARATE);
+    }
+
+    public Darknet19(int numLabels, long seed, int iterations, WorkspaceMode workspaceMode) {
+        this.numLabels = numLabels;
+        this.seed = seed;
+        this.iterations = iterations;
+        this.workspaceMode = workspaceMode;
+        this.cudnnAlgoMode = workspaceMode == WorkspaceMode.SINGLE ? ConvolutionLayer.AlgoMode.PREFER_FASTEST
+                        : ConvolutionLayer.AlgoMode.NO_WORKSPACE;
+    }
+
+    @Override
+    public String pretrainedUrl(PretrainedType pretrainedType) {
+        if (pretrainedType == PretrainedType.IMAGENET)
+            if (inputShape[1] == 448 && inputShape[2] == 448)
+                return "http://blob.deeplearning4j.org/models/darknet19_448_dl4j_inference.zip";
```

Comment:
One nit-pick here is we should add a version to the extension such as the other pretrained models. So instead this would be `http://blob.deeplearning4j.org/models/darknet19_448_dl4j_inference.v1.zip`